api/views.py

```diff
@@ -0,0 +1,176 @@
+            else:
+                first_name = employee.get().first_name
+                last_name = employee.get().last_name
+                employee.delete()
+                return Response("Employee " + first_name + " " + last_name + " fired.")
+
+    '''promotes\demotes employe based on given pesel number'''
+
+    @action(detail=False, methods=['get'])
+    def change_position(self, request, *args, **kwargs):
+        pesel = request.GET.get('pesel', None)
+        new_position = request.GET.get('new_position', None)
+
+        if pesel is not None and new_position is not None:
+            employee = Employee.objects.filter(pesel=pesel).first()
+            if not employee:
+                return Response("There is no employee with pesel: " + pesel)
+            else:
+                employee.position = new_position
+                employee.save()
+                serializer = EmployeeSerializer(employee, many=False)
+                return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def summary_cost_of_salaries(self, request, *args, **kwargs):
+        employees = Employee.objects.all()
+        sum_of_salaries = 0
+        for employee in employees:
+            sum_of_salaries += int(employee.salary)
+        return Response("Summary cost of all salaries: " + str(sum_of_salaries))
+
+
+class IncomeViewSet(viewsets.ModelViewSet):
+    queryset = Income.objects.all()
+    serializer_class = IncomeSerializer
+    filter_fields = ('name', 'date')
+
+    @action(detail=False, methods=['get'])
+    def last_incomes(self, request, *args, **kwargs):
+        number_of_incomes_to_show = request.GET.get('number_of_incomes', None)
+        if number_of_incomes_to_show is not None:
+            incomes = Income.objects.order_by('date')[:int(number_of_incomes_to_show)]
+        else:
+            incomes = Income.objects.order_by('date')
+        serializer = EventSerializer(incomes, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def sum_of_cost(self, request, *args, **kwargs):
+        days_back = request.GET.get('days', 7)
+        current_date = datetime.today().date()
+        date_for_filter = current_date - timedelta(days=int(days_back))
+        incomes = Income.objects.filter(date__gte=date_for_filter)
+        sum = 0
+        for income in incomes:
+            sum += int(income.sum)
+        return Response("Total income from the last " + str(days_back) + " days is: " + str(sum))
+
+
+class OutcomeViewSet(viewsets.ModelViewSet):
+    queryset = Outcome.objects.all()
+    serializer_class = OutcomeSerializer
+    filter_fields = ('name', 'date')
+
+    @action(detail=False, methods=['get'])
+    def last_outcomes(self, request, *args, **kwargs):
+        number_of_outcomes_to_show = request.GET.get('number_of_outcomes', None)
+        if number_of_outcomes_to_show is not None:
```

Comment:
`if number_of_outcomes_to_show` will work the same way, no reason to compare its value to `None`